tests/internal/anvil/framework.go

```diff
@@ -472,85 +472,83 @@ type ThingData struct {
 }
 
 // TestState contains client and realm data required to run a test
-type TestState interface {
-	// RealmForConfiguration returns the realm that can be used for test setup, validation and clean up
-	RealmForConfiguration() string
-	// TestRealm returns the test realm that should be passed to the IoT SDK
-	TestRealm() string
-	// Audience returns the JWT audience for the current test realm
-	Audience() string
-	// ClientType returns 'am' or 'gateway' depending on the type of client
-	ClientType() string
-	// URL of the current test server (AM or Gateway)
-	URL() *url.URL
-	// SetGatewayTree sets the auth tree used by the test IoT Gateway
-	SetGatewayTree(tree string)
+type TestState struct {
+	clientType    string
+	gateway       *gateway.Gateway
+	realm         string
+	realmPath     string
+	amURL         *url.URL
+	dnsConfigured bool
```

Comment:
I think that this is great improvement over the previous interface implementation, it is a lot cleaner.